src/gurt/telemetry.c

```diff
@@ -682,6 +707,87 @@ d_tm_build_path(char **path, char *item, va_list args)
 	return rc;
 }
 
+/**
+ * Compute gauge statistics: min, max, mean and sum of squares.
+ * Standard deviation calculation is deferred until the metric is read.
+ * This implementation computes an integer min and max with floating point
+ * mean.
+ *
+ * \param[in]	node		Pointer to a gauge node
+ */
+void
+d_tm_compute_gauge_stats(struct d_tm_node_t *node)
+{
+	struct d_tm_stats_t	*dtm_stats;
+	uint64_t		value = 0;
+	double			sum_of_squares = 0.0;
+	double			mean = 0.0;
+
+	dtm_stats = node->dtn_metric->dtm_stats;
+
+	if (dtm_stats == NULL)
+		return;
+
+	value = node->dtn_metric->dtm_data.value;
+	dtm_stats->sample_size++;
+
+	mean = dtm_stats->mean + ((value - dtm_stats->mean) /
+	       dtm_stats->sample_size);
+
+	sum_of_squares = dtm_stats->sum_of_squares +
+			 ((value - dtm_stats->mean) * (value - mean));
+
```

Comment:
I think that this involves more calculations than:
sum += val;
sum_of_squares += val * val;

Reply:
Sure, I've made the changes.  See what you think.